src/account_multisig.ts

```diff
@@ -226,14 +298,23 @@ export class Account2FA extends AccountMultisig {
             addKey(confirmOnlyKey, functionCallAccessKey(accountId, MULTISIG_CONFIRM_METHODS, null)),
             deployContract(contractBytes),
         ];
-        if ((await this.state()).code_hash === '11111111111111111111111111111111') {
-            actions.push(functionCall('new', newArgs, MULTISIG_GAS, MULTISIG_DEPOSIT),);
-        }
+        const newFunctionCallActionBatch = actions.concat(functionCall('new', newArgs, MULTISIG_GAS, MULTISIG_DEPOSIT))
         console.log('deploying multisig contract for', accountId);
-        return await super.signAndSendTransactionWithAccount(accountId, actions);
+
+        const { stateStatus: multisigStateStatus } = await this.checkMultisigCodeAndStateStatus(contractBytes);
+        switch (multisigStateStatus) {
+            case MultisigStateStatus.STATE_NOT_INITIALIZED:
+              return await super.signAndSendTransactionWithAccount(accountId, newFunctionCallActionBatch);
+            case MultisigStateStatus.VALID_STATE:
+              return await super.signAndSendTransactionWithAccount(accountId, actions);
+            case MultisigStateStatus.INVALID_STATE:
+              throw new TypedError(`Can not deploy a contract to account ${this.accountId} on network ${this.connection.networkId}, the account has existing state.`, 'ContractHasExistingState');
+            default:
+              throw new TypedError(`Can not deploy a contract to account ${this.accountId} on network ${this.connection.networkId}, the account state could not be verified.`, 'ContractStateUnknown');
+        }
     }
 
-    async disable(contractBytes: Uint8Array) {
+    async disable(contractBytes: Uint8Array, cleanupContractBytes: Uint8Array) {
```

Comment:
Please add a JSDoc on this function that references the compiled source bytes for the clean-up contract since this (and so the clean-up contract itself) is now a required parameter